addons/mail/static/src/components/message/message.js

```diff
@@ -271,28 +272,45 @@ class Message extends Component {
      * @returns {Object}
      */
     get trackingValues() {
+        /**
+         * Maps tracked field type to a JS formatter. Tracking values are not
+         * always stored in the same field type as their origin type.
+         * Field types that are not listed here are not supported by tracking in
+         * Python. Also see `create_tracking_values` in Python.
+         */
+        const typeToFormatFunctionName = {
+            boolean: 'boolean',
+            char: 'char',
+            date: 'date',
+            datetime: 'datetime',
+            float: 'float',
+            integer: 'integer',
+            /**
+             * many2one formatter exists but is expecting id/name_get or data
+             * object but only the target record name is known in this context.
+             */
+            many2one: 'char',
+            monetary: 'monetary',
+            /**
+             * Selection formatter exists but requires knowing all possibilities
+             * but they are not given in this context.
+             */
+            selection: 'char',
+            text: 'text',
+        };
         return this.message.tracking_value_ids.map(trackingValue => {
             const value = Object.assign({}, trackingValue);
             value.changed_field = _.str.sprintf(this.env._t("%s:"), value.changed_field);
-            if (value.field_type === 'datetime') {
-                if (value.old_value) {
-                    value.old_value =
-                        moment.utc(value.old_value).local().format('LLL');
-                }
-                if (value.new_value) {
-                    value.new_value =
-                        moment.utc(value.new_value).local().format('LLL');
-                }
-            } else if (value.field_type === 'date') {
-                if (value.old_value) {
-                    value.old_value =
-                        moment(value.old_value).local().format('LL');
-                }
-                if (value.new_value) {
-                    value.new_value =
-                        moment(value.new_value).local().format('LL');
-                }
+            switch (value.field_type) {
+                case 'date':
+                case 'datetime':
+                    value.old_value = value.old_value && moment.utc(value.old_value);
+                    value.new_value = value.new_value && moment.utc(value.new_value);
             }
+            const formatFunctionName = typeToFormatFunctionName[value.field_type];
+            const formatFunction = format[formatFunctionName];
+            value.old_value = formatFunction(value.old_value, undefined, { forceString: true });
+            value.new_value = formatFunction(value.new_value, undefined, { forceString: true });
```

Comment:
Avoid prop access with expression evaluation, and also passing unnecessary args to some formatters.
To do the proper way, don't define an object transformation `typeToFormatFunctionName`. Explicitly manage based on `value.field_type`. `switch`/`case` here is appropriate.

```js
switch (value.field_type) {
    case 'boolean':
        value.old_value = format.boolean(value.old_value, undefined, { forceString: true });
        value.new_value = format.boolean(value.new_value, undefined, { forceString: true });
        break;
    case 'char':
         // ...
}
```

It's longer, but the code is much more readable and safer.